util/health/health.go

```diff
@@ -587,3 +588,43 @@ func getAPIServiceHealth(obj *unstructured.Unstructured) (*appv1.HealthStatus, e
 		Message: "Waiting to be processed",
 	}, nil
 }
+
+type NodePhase string
+
+// Workflow and node statuses
+const (
+	NodePending   NodePhase = "Pending"
+	NodeRunning   NodePhase = "Running"
+	NodeSucceeded NodePhase = "Succeeded"
+	NodeSkipped   NodePhase = "Skipped"
+	NodeFailed    NodePhase = "Failed"
+	NodeError     NodePhase = "Error"
+)
+
+// An agnostic workflow object only considers Status.Phase and Status.Message. It is agnostic to the API version or any
+// other fields.
+type ApiAgnosticArgoWorkflow struct {
+	Status struct {
+		Phase NodePhase
+		Message string
+	}
+}
```

Comment:
Wasn't sure of the best place to put these. Decided on here since they are only used on the following function. Let me know if they should be moved